store/store.go

```diff
@@ -67,17 +74,25 @@ func (s *DefaultStore) SaveBlock(block *types.Block, commit *types.Commit) error
 		return fmt.Errorf("failed to marshal Commit to binary: %w", err)
 	}
 
-	bb := s.db.NewBatch()
-	err = multierr.Append(err, bb.Set(getBlockKey(hash), blockBlob))
-	err = multierr.Append(err, bb.Set(getCommitKey(hash), commitBlob))
-	err = multierr.Append(err, bb.Set(getIndexKey(block.Header.Height), hash[:]))
+	badgerDS, ok := s.db.(*badger3.Datastore)
+	if !ok {
+		return errors.New("failed to retrieve the ds.Datastore implementation")
+	}
+	bb, err := badgerDS.NewTransaction(s.ctx, false)
```

Comment:
the parameter refers to is-read-only, hence by passing false, we want this new transaction to be write enabled.

Reply:
> The first argument to DB.NewTransaction() is a boolean stating if the transaction should be writable.

In my understanding, this means that false is "should not be writable" -> If we want to make it write enabled then this should be true and we should use `Txn.Commit` instead of `Txn.Discard` .  
Source: https://dgraph.io/docs/badger/get-started/#managing-transactions-manually

Reply:
- the parameter refers to `readOnly`, hence passing `false` makes it writable. `true` will make it non-updatable.
- we are calling `Commit` on success, we only do `Discard` on error case.
let me know if it is not clear.

Reply:
Ok Thank you for clarifying, looking into the Code would have been less confusing than reading the documentation. 
This is clarified now. One thing is still unclear to me. Why do we need write access, do we ever change the Txn?

Reply:
yes, we update the Txn for putting entries into it: https://github.com/dgraph-io/badger/blob/release/v3.2011/txn.go#L419
the code is basically creating an empty NewTransaction and then populating it.

Reply:
Ok that makes sense now ty :)